src/lib/features/client-feature-toggles/tests/client-feature-toggles-auth.e2e.test.ts

```diff
@@ -0,0 +1,59 @@
+import dbInit, { ITestDb } from '../../../../test/e2e/helpers/database-init';
+import {
+    IUnleashTest,
+    setupAppWithAuth,
+} from '../../../../test/e2e/helpers/test-helper';
+import getLogger from '../../../../test/fixtures/no-logger';
+import { IApiToken, ApiTokenType } from '../../../types/models/api-token';
+
+let app: IUnleashTest;
+let db: ITestDb;
+let clientToken: IApiToken;
+
+beforeAll(async () => {
+    db = await dbInit('client_api_auth', getLogger);
+    app = await setupAppWithAuth(
+        db.stores,
+        {
+            frontendApiOrigins: ['https://example.com'],
+        },
+        db.rawDatabase,
+    );
+
+    clientToken = await app.services.apiTokenService.createApiTokenWithProjects(
+        {
+            type: ApiTokenType.CLIENT,
+            projects: ['*'],
+            environment: 'default',
+            tokenName: `client-api-auth`,
+        },
+    );
+});
+
+afterAll(async () => {
+    await app.destroy();
+    await db.destroy();
+});
+
+test('should fail requests without auth headers', async () => {
+    await app.request
+        .get('/api/client/features')
+        .expect('Content-Type', /json/)
+        .expect(401);
+});
+
+test('should allow requests to /api/client/features with authorization header', async () => {
+    await app.request
+        .get('/api/client/features')
+        .set('authorization', clientToken.secret)
+        .expect('Content-Type', /json/)
+        .expect(200);
+});
+
+test('should allow requests to /api/client/features with x-unleash-auth header', async () => {
+    await app.request
+        .get('/api/client/features')
+        .set('x-unleash-auth', clientToken.secret)
+        .expect('Content-Type', /json/)
+        .expect(200);
+});
```

Comment:
One edge case to consider:
If we send both headers, which one takes precedence? I believe we should read from `x-unleash-auth` and don't fail if both are present. Maybe we can test it's still 200 when x-unleash-auth has a valid token, having also the Authorization header with an invalid string.

Reply:
Agree. That should be the default behavior, I'll add a test that verifies it.

Reply:
addressed in [this commit](https://github.com/Unleash/unleash/pull/5568/commits/fccbd18a042cf990b3dfa4102f91a48fe8205d1a)